cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -154,6 +154,18 @@
     # adding a missing bpf mount
     mount fstype=bpf options=(rw) bpf -> /sys/fs/bpf/,
 
+    # For mounting base dir by dir (write dirs and mount on them)
+    /tmp/snap.rootfs_** rw,
+    mount options=(remount ro) -> /tmp/snap.rootfs_*/,
+    mount options=(rw rbind) {/var/lib/snapd,}/snap/core**/ -> /tmp/snap.rootfs_**/,
+    # For mounting individual files
+    mount options=(rw bind) {/var/lib/snapd,}/snap/core** -> /tmp/snap.rootfs_**,
+    mount options=(rw rslave) -> /tmp/snap.rootfs_**/,
+    # For mounting a specific base snap
+    mount options=(rw rbind) @SNAP_MOUNT_DIR@/*/*/*/ -> /tmp/snap.rootfs_*/*/,
```

Comment:
I am not sure why this one is needed. Is not it included by the rules before?

Reply:
I removed it and I get errors (from https://github.com/snapcore/snapd/actions/runs/3173187723/jobs/5168811507): 
```
2022-10-03T10:56:09.1271972Z + test-snapd-busybox-static.busybox-static echo hello
2022-10-03T10:56:09.1272830Z cannot perform operation: mount --rbind /var/lib/snapd/snap/bare/current/lib /tmp/snap.rootfs_ICd7Al/lib: Permission denied
```

Reply:
Would that be enough?

```
/tmp/snap.rootfs_** rw,
mount options=(remount ro) -> /tmp/snap.rootfs_*/,
mount options=(rw rbind)  @SNAP_MOUNT_DIR@/*/*/**/ -> /tmp/snap.rootfs_**/,
mount options=(rw bind)  @SNAP_MOUNT_DIR@/*/*/** -> /tmp/snap.rootfs_*/**,
mount options=(rw rslave) -> /tmp/snap.rootfs_**/,
mount options=(rw rbind) /*/ -> /tmp/snap.rootfs_**/,
```

Reply:
This seems to work, thanks! I've pushed a new commit using this set of rules.